pkg/controllers/placement/namespace_status_sync.go

```diff
@@ -27,81 +27,268 @@ import (
+			Message:            msg,
+			ObservedGeneration: generation,
+		}
+	}
+
+	if isResourceSelectorInvalid {
+		return metav1.Condition{
+			Type:               string(placementv1beta1.ClusterResourcePlacementScheduledConditionType),
+			Status:             metav1.ConditionFalse,
+			Reason:             condition.InvalidResourceSelectorsReason,
+			Message:            msg,
+			ObservedGeneration: generation,
+		}
+	}
+
+	return metav1.Condition{
+		Type:               string(placementv1beta1.ClusterResourcePlacementStatusSyncedConditionType),
+		Status:             metav1.ConditionTrue,
+		Reason:             condition.StatusSyncSucceededReason,
+		Message:            msg,
+		ObservedGeneration: generation,
 	}
+}
 
-	// Extract target namespace.
-	targetNamespace := extractNamespaceFromResourceSelectors(placementObj)
+// setConditionAndUpdateStatus is a helper function that sets the argued condition on the placement
+// object and updates its status.
+func (r *Reconciler) setConditionAndUpdateStatus(ctx context.Context, placementObj placementv1beta1.PlacementObj, condition metav1.Condition) error {
+	placementKObj := klog.KObj(placementObj)
+
+	// Set the condition on the placement object.
+	placementObj.SetConditions(condition)
+
+	// Update the placement status.
+	if err := r.Client.Status().Update(ctx, placementObj); err != nil {
+		klog.ErrorS(err, "Failed to update the placement status with StatusSynced condition", "crp", placementKObj)
+		return controller.NewUpdateIgnoreConflictError(err)
+	}
+
+	klog.V(2).InfoS("Updated the placement status with StatusSynced condition", "crp", placementKObj)
+
+	return nil
+}
+
+// syncClusterResourcePlacementStatus creates or updates a ClusterResourcePlacementStatus
+// object in the target namespace for ClusterResourcePlacements with NamespaceAccessible scope.
+// It extracts the target namespace from the CRP's resource selectors, creates/updates a CRPS object
+// with filtered status (excluding StatusSynced condition), and sets the CRP as owner for
+// automatic cleanup. Returns an error if the operation fails or if the target namespace name is empty.
+func (r *Reconciler) syncClusterResourcePlacementStatus(ctx context.Context, placementObj placementv1beta1.PlacementObj) (string, error) {
+	placementKObj := klog.KObj(placementObj)
+	crp, _ := placementObj.(*placementv1beta1.ClusterResourcePlacement)
+
+	// Extract target namespace from resource selectors.
+	targetNamespace := extractNamespaceFromResourceSelectors(crp)
 	if targetNamespace == "" {
-		// Not NamespaceAccessible or no namespace found - skip sync.
-		klog.V(2).InfoS("Skipped processing CRP to create/update ClusterResourcePlacementStatus", "crp", klog.KObj(crp))
-		return nil
+		klog.ErrorS(controller.NewUnexpectedBehaviorError(fmt.Errorf("no namespace selector found despite NamespaceAccessible scope")),
+			"Failed to find valid Namespace selector for NamespaceAccessible scope",
+			"crp", placementKObj)
+		return "", nil
```

Comment:
Nit: The fact that we are not returning error here is a bit odd. I know the reason is that the buildNamespaceAccessibleCondition function takes in two bool (isSyncError, isResourceSelectorInvalid) but conceptually, it should  only take in one. The function already takes in the msg but choose to decide the reason field.  This awkward setup could use some streamline